app/src/main/java/com/capstone/weatherapp/WeatherRepository.kt

```diff
@@ -26,23 +26,40 @@ class WeatherRepository(private val db: WeatherDatabase) {
 
     suspend fun refreshCityList() {
         withContext(Dispatchers.IO) {
+            _cityListData.postValue(State.Loading)
             try {
-                val cityList = WeatherApiClient.retrofitService.getCities(CITIES, APIKEY, UNITS)
-                db.cityDao().insert(cityList.convertToCityCacheList())
-            } catch (e: Exception) {}
-            _cityListData.postValue(getNetworkCityList())
+                val cityListResponse = WeatherApiClient.retrofitService.getCities(CITIES, APIKEY, UNITS)
+                if (cityListResponse.isSuccessful) {
```

Comment:
you shouldn't need this if statement here, if it isn't successful it will throw an exception which is what the catch block is for